Source/WebCore/PAL/pal/text/TextEncoding.h

```diff
@@ -61,7 +62,7 @@ class TextEncoding : public WTF::URLTextEncoding {
     bool isNonByteBasedEncoding() const;
     bool isUTF7Encoding() const;
 
-    ASCIILiteral m_name;
+    AtomString m_name;
```

Comment:
Seems like a regression? The previous code never did any allocations since it was using ASCIILiteral, now we have a heap allocated object instead.

Reply:
Something's not right, I think. These paths were hitting the StringImpl constructor with the `"UTF-8"` string. I can back this out and grab a trace showing what that looked like. Maybe I misunderstood what was going on?

Reply:
Yes, I'm curious where the StringImpl construction was from. ASCIILiteral is just a literal `const char*`, not String/StringImpl involved.